src/main/java/org/jboss/pnc/deliverablesanalyzer/Finder.java

```diff
@@ -192,33 +181,44 @@ private void initCaches(BuildConfig config) {
                             krbCCache,
                             krbKeytab)
                     : new KojiClientSession(config.getKojiHubURL())) {
+                LOGGER.info("Initialized Koji client session with URL {}", config.getKojiHubURL());
+
                 BuildFinder finder;
 
                 if (config.getPncURL() != null) {
-                    PncClient14 pncclient = new PncClient14(config);
+                    PncClient pncclient = new HashMapCachingPncClient(config);
+                    LOGGER.info("Initialized PNC client with URL {}", config.getPncURL());
                     finder = new BuildFinder(session, config, analyzer, cacheManager, pncclient);
                 } else {
                     finder = new BuildFinder(session, config, analyzer, cacheManager);
                 }
 
+                LOGGER.info("Initialized finder");
+
                 Future<Map<BuildSystemInteger, KojiBuild>> futureBuilds = pool.submit(finder);
 
                 try {
                     checksums = futureChecksum.get();
                     builds = futureBuilds.get();
 
-                    if (LOGGER.isDebugEnabled()) {
-                        LOGGER.debug("Got {} checksums and {} builds", checksums.size(), checksums.size());
-                    }
+                    LOGGER.info("Got {} checksums and {} builds", checksums.size(), builds.size());
                 } catch (ExecutionException e) {
+                    LOGGER.error("Got ExecutionException", e);
                     // throw new KojiClientException("", e);
                 } catch (InterruptedException e) {
                     Thread.currentThread().interrupt();
                 }
+
+                FinderResult result = new FinderResult(builds, username);
+
+                LOGGER.info("Initialized finder result");
+
+                return result;
             } catch (KojiClientException e) {
-                // ignore
+                LOGGER.error("Got KojiClientException", e);
```

Comment:
The exceptions handling should be reworked to fulfill these requirements:
- It must be easily readable (some of the catch clauses should be in a series as catch (ex1) {} catch (ex2) as it mainly just logs the errors
- If smth goes wrong a readable error must be passed to the users. Currently it just silently ignores it

Reply:
I think you're talking about the nested exceptions right? I will either unnest them or move them to a separate method as part of refactoring this method.

When I try to throw an exception up to the endpoint, it just stops working. That's why they are just logged right now. This really needs to be solved.

Reply:
OK, the endpoint catches the exceptions and re-throws a `javax.ws.rs.WebApplicationException`.

Reply:
The problem is that the exception doesn't seem to get handled by `org.jboss.pnc.deliverablesanalyzer.rest.ErrorMapper` so that the exception message is `null`.

Right now, I am just logging and re-throwing the exception so that we can at least see the exception in the log.

But, the exception is not making it to the client.